src/trc.py

```diff
@@ -195,32 +200,59 @@ def _import_from_c3d(self, filename):
             self['CameraRate'] = reader.header.frame_rate
-            self['NumFrames'] = reader.header.last_frame
-            self['NumMarkers'] = reader.get('POINT').get('USED').int16_value
+            self['NumFrames'] = reader.header.last_frame - reader.header.first_frame + 1
             self['Units'] = reader.get('POINT').get('UNITS').string_value
             self['OrigDataRate'] = reader.header.frame_rate
             self['OrigDataStartFrame'] = reader.header.first_frame
             self['OrigNumFrames'] = reader.header.last_frame - reader.header.first_frame + 1
 
-            # Set data column labels.
-            self['Markers'] = [label.strip() for label in reader.point_labels]
-
             # Set frame numbers.
             self['Frame#'] = [i for i in range(reader.header.first_frame, reader.header.last_frame + 1)]
 
+            if filter_output is None:
+                filter_output = ['ANGLES', 'FORCES', 'MOMENTS', 'POWERS', 'SCALARS']
+            if label_params is None:
+                label_params = ['LABELS', 'LABELS2']
+
+            # Filter out model outputs (Angles, Forces, Moments, Powers, Scalars) from point labels.
+            point_group = reader.get('POINT')
+            model_outputs = set()
+            for param in filter_output:
+                if param in point_group.param_keys():
+                    model_outputs.update(point_group.get(param).string_array)
+            point_labels = []
+            for param in label_params:
+                if param in point_group.param_keys():
+                    filtered_labels = [None if label in model_outputs else label.strip() for label in point_group.get(param).string_array]
+                    point_labels.extend(filtered_labels)
+
+            # Set marker labels.
+            self['Markers'] = [label for label in point_labels if label]
+            self['NumMarkers'] = len(self['Markers'])
+
             # Set marker data.
             for i, points, analog in reader.read_frames():
                 time = (i - 1) * (1 / reader.point_rate)
-                self[i] = time, [point[:3].tolist() for point in points]
-
-    def import_from(self, filename):
+                point_data = []
+                for j in range(len(points)):
+                    if point_labels[j]:
+                        coordinates = points[j][:3].tolist()
+                        errors = points[j][3:]
+                        for error in errors:
+                            if error == -1:
+                                coordinates = _convert_coordinates(['', '', ''])
+                                break
+                        point_data.append(coordinates)
+                self[i] = time, point_data
+
+    def import_from(self, filename, *args, **kwargs):
         """
         Import data from a non-TRC file source.
         Currently, the only alternative supported format is c3d.
 
         :param filename: The source file of the data to be imported.
```

Comment:
Just need to document those keyword arguments here that you can now use, fancy.

Reply:
I've just added that documentation. Let me know if you want me to add/remove/change anything.